src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -307,11 +310,37 @@ private void evaluateSystemIndicesAccess(
                         );
                     }
                     presponse.allowed = false;
+                    presponse.addMissingPrivileges(action);
                     presponse.markComplete();
                     return;
                 }
         }
 
+        if (user.isPluginUser()) {
```

Comment:
I think this distinction should be kept here since this is an "ownership" model. Authorization is done based on identity here and not a role. i.e. PluginA registers SystemIndexA and gets full access to it by default. 

We have a mechanism for provisioning system index access to regular user's but it relies on a feature being enabled. The feature flag is `plugins.security.system_indices.permission.enabled` and its off by default.

Reply:
> Authorization is done based on identity here and not a role. i.e. PluginA registers SystemIndexA and gets full access to it by default.

Why add an alternative authorization model for this case?  Seems like this makes the system more complex to understand and verify vs role checks.

Reply:
@nibix @DarshitChanpura I've got a preference for consistent treatment Users objects no matter if they represent a person or a plugin, I'd be curious for your opinions one way or the other.

Reply:
@peternied For what its worth, there is [service account specific logic](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java#L260-L277) in the same file. Service account is analogous to plugin user, but the difference is that its meant for extensions. i.e. If there are extensions in a cluster that register system indices, then on bootstrap the security plugin issues a token to the extension that it can use to access its registered system indices if it wants to use OpenSearch as a datastore.

Reply:
From what I understand, we are trying to introduce a concept of Plugin Awareness. This requires Identification of the plugin. To enforce this identity we are leveraging already existing User authz flow. Each plugin should have access to its own resource by default, but no other indices. We already have "permission-types" in place, so IMO introducing a plugin user is manageable. We can mark plugin specific users and roles as reserved and static (probly hidden?) to prevent them from being unintentionally modified.

Reply:
> We can mark plugin specific users and roles as reserved and static (probly hidden?) to prevent them from being unintentionally modified.

These aren't tied to an index. They are purely in-memory and cannot be altered.